app/config/collections.php

```diff
@@ -1366,7 +1377,495 @@
+                'orders' => [Database::ORDER_ASC],
+            ],
+            [
+                '$id' => ID::custom('_key_providerInternalId'),
+                'type' => Database::INDEX_KEY,
+                'attributes' => ['providerInternalId'],
+                'lengths' => [128],
+                'orders' => [Database::ORDER_ASC],
+            ],
+            [
+                '$id' => ID::custom('_key_search'),
+                'type' => Database::INDEX_FULLTEXT,
+                'attributes' => ['search'],
+                'lengths' => [],
+                'orders' => [],
+            ],
+        ],
+    ],
+
+    'topics' => [
+        '$collection' => ID::custom(DATABASE::METADATA),
+        '$id' => ID::custom('topics'),
+        'name' => 'Topics',
+        'attributes' => [
+            [
+                '$id' => ID::custom('providerId'),
+                'type' => Database::VAR_STRING,
+                'format' => '',
+                'size' => Database::LENGTH_KEY,
+                'signed' => true,
+                'required' => true,
+                'default' => null,
+                'array' => false,
+                'filters' => [],
+            ],
+            [
+                '$id' => ID::custom('providerInternalId'),
+                'type' => Database::VAR_STRING,
+                'format' => '',
+                'size' => Database::LENGTH_KEY,
+                'signed' => true,
+                'required' => true,
+                'default' => null,
+                'array' => false,
+                'filters' => [],
+            ],
+            [
+                '$id' => ID::custom('name'),
+                'type' => Database::VAR_STRING,
+                'format' => '',
+                'size' => 128,
+                'signed' => true,
+                'required' => true,
+                'default' => null,
+                'array' => false,
+                'filters' => [],
+            ],
+            [
+                '$id' => ID::custom('description'),
+                'type' => Database::VAR_STRING,
+                'format' => '',
+                'size' => 2048,
+                'signed' => true,
+                'required' => false,
+                'default' => null,
+                'array' => false,
+                'filters' => [],
+            ],
+            [
+                '$id' => ID::custom('targets'),
+                'type' => Database::VAR_STRING,
+                'format' => '',
+                'size' => 16384,
+                'signed' => true,
+                'required' => false,
+                'default' => null,
+                'array' => false,
+                'filters' => ['subQueryTopicTargets'],
+            ]
+        ],
+        'indexes' => [
+            [
+                '$id' => ID::custom('_key_providerId'),
+                'type' => Database::INDEX_KEY,
+                'attributes' => ['providerId'],
+                'lengths' => [128],
```

Comment:
We can leave the index length empty since attribute _key_providerId is 128 in length. 
We do not exceed the length max.

Reply:
done this